dgp/utils/render_3d_to_2d.py

```diff
@@ -0,0 +1,422 @@
+        a dictionary with key is the camera name, value is a dictionary whose key is class_name of bounding_box_2d,
+        value is list of Tuple [bounding_box_2d, bounding_box_3d, centroid_2d]
+    """
+    bboxes_2d_from_lidars = render_bounding_boxes_3d_of_lidars(
+        dataset, camera_datum_names, lidar_datum_names, max_num_items
+    )
+    bboxes_2d_from_cameras = render_bounding_boxes_2d_of_cameras(dataset, camera_datum_names, max_num_items)
+    associated_bboxes = defaultdict(lambda: defaultdict(list))
+    for camera_name in camera_datum_names:
+        bboxes_2d_from_lidar = bboxes_2d_from_lidars[camera_name]
+        bboxes_2d_from_camera = bboxes_2d_from_cameras[camera_name]
+        for name_2d, name_3d in ontology_name_mapper.items():
+            logging.info("{}: Associate {} to {}".format(camera_name, name_2d, name_3d))
+            bboxes_from_lidar = None
+            if name_3d in bboxes_2d_from_lidar:
+                bboxes_from_lidar = bboxes_2d_from_lidar[name_3d]
+            bboxes_from_camera = None
+            if name_2d in bboxes_2d_from_camera:
+                bboxes_from_camera = bboxes_2d_from_camera[name_2d]
+            associated_bboxes[camera_name][name_2d] = associate_lidar_and_camera_2d_bboxes(
+                bboxes_from_camera, bboxes_from_lidar
+            )
+    return associated_bboxes
+
+
+def associate_3d_and_2d_annotations_scene(
+    scene_json: str, ontology_name_mapper: dict, camera_datum_names: Optional[list[str]],
+    lidar_datum_names: Optional[list[str]], max_num_items: Optional[int]
+) -> defaultdict[defaultdict[list[Tuple]]]:
+    """Associate 3d bounding boxes and 2d bounding boxes to the same object with given scene.
+    Parameters
+    ----------
+    scene_json: str
+        Full path to the scene json.
+    ontology_name_mapper: dict
+        Map the class names from bounding_box_2d to bounding_box_3d if the class names are different.
+        eg: {'Pedestrian': 'Person','Car': 'Car'}.
+    camera_datum_names: Optional[list[str]]
+        List of camera names.
+        If None, use all the cameras available in the DGP dataset.
+    lidar_datum_names: Optional[list[str]]
+        List of lidar names.
+        If None, use all the lidars available in the DGP dataset.
+    max_num_items: Optional[int]
+        If not None, then show only up to this number of items. This is useful for debugging a large dataset.
+        Default: None.
+    Returns
+    -------
+    associated_bboxes: defaultdict[defaultdict[list[Tuple]]]
+        a dictionary with key is the camera name, value is a dictionary whose key is class_name of bounding_box_2d,
+        value is list of Tuple [bounding_box_2d, bounding_box_3d, centroid_2d]
+    """
+    datum_names = camera_datum_names + lidar_datum_names
+    dataset = SynchronizedScene(
+        scene_json,
+        datum_names=datum_names,
+        requested_annotations=[ANNOTATIONS_2D, ANNOTATIONS_3D],
+        only_annotated_datums=True
```

Comment:
nit: only_annotated_datums means annotations_2d or annotation_3d, it does not guarantee you will have both